### PR TITLE
Fix some block comments in Rational

### LINKED ISSUE #2930: block comment in Rational between lparen and numerator is moved before lparen
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/main?data=N4KABGBEDGD2AmBTSAuKAmAzAFgDwGsBzAPQApSAqMAM1ljAoEp0B6TRgPkgBpwoBnRABchASwB2hfqjABtPhFARlUACRCAngAdkaSBKE8FyyKoBuAQwA2AV11zjKgAy8VKyAElxScUIDKogBeyK5uENiOYAC6xgC%2BoYqRppo6Mvq%2BRmGmlrb28mEQAIwJbpAAshYAHgAyEojViJJCABaZBYXoTpExKvHGSqXq2vaQjfAA8tS14oh%2BmlYhSebWdjL5Begl7gCi3pPTiwVQVtSQ3XElA%2B5DqXoARnRWbdc5q2jrYZhbJl6CAE5CABiEmsADlEAB3Kx1Z5uIR-OznXqXJYpEYPWBPb7ZFZ5SLhb4QSB%2BLQWaCIABCiFof0QAAULH8LABbYSIP6wlTwxFhHrKPoqK4mG7ox6ctSvPFHACshKgJLJlOpsFp1VgEPZ0Asgi8ZlgWrEsHE4og3MQSP5KKyIrSGKxS0la3xYAAbHLiaTyVSaYgAKpaHR-LU68R6g2iI0mmjWQQWiAC5RCok2%2B5i7HLXJOo4AdndCq9ytpAGErNr%2BEWjfxudAhCqo9QY%2BbeRd%2Bqjhra0w7cVmCgAOPOepU%2BsqIZl3dn1xtxsAJxLWtEdzHinGZ97OgCcA8V3pViArVkjcobVljzeRrfn7dTS-TjrXR0KLmdHsVAEFqEJ2RXmcyLFGzdOs5gEmagLte9rWneDgPsUz75kOu5%2BKOohwAexpHlOZ6WhegxgVAdrLhmbzQe0mxwYO76fn8SHMihmKHs6AFYfGVq4Ve%2BGdpB3b3u0XzkW%2BfywDY3gACKINCtFUf%2BCJNm4fIsTh1x4ekhi3txJFhIU2DuhUlQeNQAAqzSNH4zQqkIADqojwC0UZdMxM6sUp7EqYRUEfG4hSys%2Bun6UZjTbCesxmQCVk2a0couvZcktoKba3FABhuepHkqIUbo%2BVUXjUKIlTjIGFi1n82yVFotL8PwEboc6vbRSo8mOYpwrKUlamrhpnm5pllQAEqIHAfzwGFtlytgdXKA1QEgckLmtV27WpcohT9t1fUDfAoI2GO7KTB4n7MtIcqFIBTnNS5zI2FYYjQjMgIqr%2BIjsgZ7ZtcRi1FJuz5rSq8BlJd111HdfwPVJ7rQM0jJklRAD6ELWbZJ1NcmLUZK9UobE%2BRzlFUr5-EyGjjH8tRVsNEU1eNECTadyOzaj81vc66CwVjum4-jhPE0Im3bX8u37YdzrHQ5U3xSMF1XaIN2IEDINPS99Po2E6BkVjbMWATROiFWf0S1LMuFaDz7g5DNbsrD8OtIjcWXglrloz2St8SzVQAGq4hSEjwBIhCk1GtVW4motpHNXELYz2ndYCwk1lVHveN7vujRT0SxYHNsjCHgzuYz3nO5UImwEIADiwglWViAVVVifkwHc5sbbBH2zxSsZVj4JQnUVJCBqjTPToYk5eIoiGuIr7eCOPMC0cTExee1v16KN4Kw7bjoF1qvQoQ4hR%2BIMdGgEW%2BFTYtIGbAXg%2BEIhVVZOQW18BQfgclYdHOgK0b1R4hX2Y9RGoQE-jn8AeEhh5VSngUY8p5Z7YXns5BunEs4pUZp9Fm-1RAHGqCyO48ALAllgJVSQ7cpY30gfVVOddYGLwggg5%2BBRMCYwKJAMu7JRCsl8NYAA0ogRAWhz6NCEF4CkTJd5kyOBA2SpC55pwXouKhLxEFHEwMzBhFJSziHwAcfguMhLeHBFWRAv1UFSzLuVSqlZpI8igQpGBZ04FL1DgzBRKtlGMh3LSESWtoB-BYSCT88BfRDyNGJaApYmQj2IeIiaZD77pzSOLAGMxBFknwMIOYGgFhPwcbQp2DCdbxMpEyaAyT-DzEOAwzxLIdDwDOMLamoFaaqWXs3NwOB3ScO4bpbmADJgqIsGojRUZHzJypkjOptjZHCmzgo3ODDCGd0LIgXJks6jfi0DYS%2BI9jEV1MdVaeMk77TRTBxOx1DMmfFbgwphXjWGXysIFWi-ARHgMwpYmcfAoihH0PwQElUZBiJALEIAA)

#### Code

```fsharp
234<kg^((* foo *)2/3)>
```
            

#### Result

```fsharp
234<kg^ (* foo *) (2/3)>

```
            
#### Problem description

Please describe here the Fantomas problem you encountered.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/main/CONTRIBUTING.md#bug-reports).

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.
- [ ] I would like a release if this problem is solved.

#### Options

Fantomas main branch at 2023-07-24T13:06:38Z - 0e07caba6301511d70a8187d5b6a1982d6c98638

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>
<sub>PS: It's unlikely that someone else will solve your specific issue, as it's something that you have a personal stake in.</sub>

### LINKED ISSUE #2932: block comment in Rational between / and denominator is moved before /
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/main?data=N4KABGBEDGD2AmBTSAuKAmAzAFgDwGsBzAPQAp0B6UgKjADNZYxqBKTFgPkgBpwoBnRABchASwB2hfqjABtPhFARlUACRCAngAdkaSBKE8FyyKoBuAQwA2AV11zjKgAy8VKyAElxScUIDKogBeyK5uENiOYAC6xgC%2BoYqRppo6Mvq%2BRmGmlrb28mEQAIwJbpAAshYAHgAyEojViJJCABaZBYXoTpExKvHGSqXq2vaQjfAA8nS14oh%2BmlYhSebWdjL5Begl7gCi3pPTiwVQVnSQ3XElA%2B5DqXoARoxWbdc5q2jrYZhbJl6CAE5CABiEmsADlEAB3Kx1Z5uIR-OznXqXJYpEYPWBPb7ZFZ5SLhb4QSB%2BLQWaCIABCiAYf0QAAULH8LABbYSIP6wlTwxFhHrKPoqK4mG7ox6ctSvPFHACshKgJLJlOpsFp1VgEPZ0Asgi8ZlgWrEsHE4og3MQSP5KKyIrSGKxS0la3xYAAbHLiaTyVSaYgAKpaHR-LU68R6g2iI0m%2BjWQQWiAC5RCok2%2B5i7HLXJOo4AdndCq9ytpAGErNr%2BEWjfxudAhCqo3QY%2BbeRd%2Bqjhra0w7cVmCgAOPOepU%2BsqIZl3dn1xtxsAJxLWtEdzHinGZ97OgCcA8V3pViArVkjcobVljzeRrfn7dTS-TjrXR0KLmdHsVAEE6EJ2RXmcyLFGzdOs5gEmagLte9rWneDgPsUz75kOu5%2BKOohwAexpHlOZ6WhegxgVAdrLhmbzQe0mxwYO76fn8SHMihmKHs6AFYfGVq4Ve%2BGdpB3b3u0XzkW%2BfywDY3gACKINCtFUf%2BCJNm4fIsTh1x4ekhi3txJFhIU2DuhUlQeHQAAqzSNH4zQqkIADqojwC0UZdMxM6sUp7EqYRUEfG4hSys%2Bun6UZjTbCesxmQCVk2a0couvZcktoKba3FABhuepHkqIUbo%2BVUXh0KIlTjIGFi1n82yVFotL8PwEboc6vbRSo8mOYpwrKUlamrhpnm5pllQAEqIHAfzwGFtlytgdXKA1QEgckLmtV27WpcohT9t1fUDfAoI2GO7KTB4n7MtIcqFIBTnNS5zI2FYYjQjMgIqr%2BIjsgZ7ZtcRi1FJuz5rSq8BlJd111HdfwPVJ7rQM0jJklRAD6ELWbZJ1NcmLUZK9UobE%2BRzlFUr5-EyGjjH8tRVsNEU1eNECTadyOzaj81vc66CwVjum4-jhPE0Im3bX8u37YdzrHQ5U3xSMF1XaIN2IEDINPS99Po2E6BkVjbMWATROiFWf0S1LMuFaDz7g5DNbsrD8OtIjcWXglrloz2St8SzVQAGq4hSEjwBIhCk1GtVW4motpHNXELYz2ndYCwk1lVHveN7vujRT0SxYHNsjCHgzuYz3nO5UImwEIADiwglWViAVVVifkwHc5sbbBH2zxSsZVj4JQnUVJCBqjTPToYk5eIoiGuIr7eCOPMC0cTExee1v16KN4Kw7bjoF1qvQoQ4hR%2BIMdGgEW%2BFTYtIGbAXg%2BEIhVVZOQW18BQfgclYdHOgK0b1R4hX2Y9RGoQE-jn8AeEhh5VSngUY8p5Z7YXns5BunEs4pUZp9Fm-1RAHGqCyO48ALAllgJVSQ7cpY30gfVVOddYGLwggg5%2BBRMCYwKJAMu7JRCsl8NYAA0ogRAWhz6NCEF4CkTJd5kyOBA2SpC55pwXouKhLxEFHEwMzBhFJSziHwAcfguMhLeHBFWRAv1UFSzLuVSqlZpI8igQpGBZ04FL1DgzBRKtlGMh3LSESWtoB-BYSCT88BfRDyNGJaApYmQj2IeIiaZD77pzSOLAGMxBFknwMIOYGgFhPwcbQp2DCdbxMpEyaAyT-DzEOAwzxLIdDwDOMLamoFaaqWXs3NwOB3ScO4bpbmADJgqIsGojRUZHzJypkjOptjZHCmzgo3ODDCGd0LIgXJks6jfi0DYS%2BI9jEV1MdVaeMk77TRTBxOx1DMmfFbgwphXjWGXysIFWi-ARHgMwpYmcfAoihH0PwQElUZBiJALEIAA)

#### Code

```fsharp
234<kg^(2/(* foo *)3)>
```
            

#### Result

```fsharp
234<kg^(2 (* foo *) /3)>
```
            
#### Problem description

Please describe here the Fantomas problem you encountered.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/main/CONTRIBUTING.md#bug-reports).

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.
- [ ] I would like a release if this problem is solved.

#### Options

Fantomas main branch at 2023-07-24T13:06:38Z - 0e07caba6301511d70a8187d5b6a1982d6c98638

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>
<sub>PS: It's unlikely that someone else will solve your specific issue, as it's something that you have a personal stake in.</sub>

### LINKED ISSUE #2933: block comment in Rational between denominator and rparen is moved behind rparen
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/main?data=N4KABGBEDGD2AmBTSAuKAmAzAFgDwGsBzAPQAp0B6TUgKjADNZYwaBKVgPkgBpwoBnRABchASwB2hfqjABtPhFARlUACRCAngAdkaSBKE8FyyKoBuAQwA2AV11zjKgAy8VKyAElxScUIDKogBeyK5uENiOYAC6xgC%2BoYqRppo6Mvq%2BRmGmlrb28mEQAIwJbpAAshYAHgAyEojViJJCABaZBYXoTpExKvHGSqXq2vaQjfAA8vS14oh%2BmlYhSebWdjL5Begl7gCi3pPTiwVQVvSQ3XElA%2B5DqXoARkxWbdc5q2jrYZhbJl6CAE5CABiEmsADlEAB3Kx1Z5uIR-OznXqXJYpEYPWBPb7ZFZ5SLhb4QSB%2BLQWaCIABCiEYf0QAAULH8LABbYSIP6wlTwxFhHrKPoqK4mG7ox6ctSvPFHACshKgJLJlOpsFp1VgEPZ0Asgi8ZlgWrEsHE4og3MQSP5KKyIrSGKxS0la3xYAAbHLiaTyVSaYgAKpaHR-LU68R6g2iI0mhjWQQWiAC5RCok2%2B5i7HLXJOo4AdndCq9ytpAGErNr%2BEWjfxudAhCqo-QY%2BbeRd%2Bqjhra0w7cVmCgAOPOepU%2BsqIZl3dn1xtxsAJxLWtEdzHinGZ97OgCcA8V3pViArVkjcobVljzeRrfn7dTS-TjrXR0KLmdHsVAEF6EJ2RXmcyLFGzdOs5gEmagLte9rWneDgPsUz75kOu5%2BKOohwAexpHlOZ6WhegxgVAdrLhmbzQe0mxwYO76fn8SHMihmKHs6AFYfGVq4Ve%2BGdpB3b3u0XzkW%2BfywDY3gACKINCtFUf%2BCJNm4fIsTh1x4ekhi3txJFhIU2DuhUlQePQAAqzSNH4zQqkIADqojwC0UZdMxM6sUp7EqYRUEfG4hSys%2Bun6UZjTbCesxmQCVk2a0couvZcktoKba3FABhuepHkqIUbo%2BVUXj0KIlTjIGFi1n82yVFotL8PwEboc6vbRSo8mOYpwrKUlamrhpnm5pllQAEqIHAfzwGFtlytgdXKA1QEgckLmtV27WpcohT9t1fUDfAoI2GO7KTB4n7MtIcqFIBTnNS5zI2FYYjQjMgIqr%2BIjsgZ7ZtcRi1FJuz5rSq8BlJd111HdfwPVJ7rQM0jJklRAD6ELWbZJ1NcmLUZK9UobE%2BRzlFUr5-EyGjjH8tRVsNEU1eNECTadyOzaj81vc66CwVjum4-jhPE0Im3bX8u37YdzrHQ5U3xSMF1XaIN2IEDINPS99Po2E6BkVjbMWATROiFWf0S1LMuFaDz7g5DNbsrD8OtIjcWXglrloz2St8SzVQAGq4hSEjwBIhCk1GtVW4motpHNXELYz2ndYCwk1lVHveN7vujRT0SxYHNsjCHgzuYz3nO5UImwEIADiwglWViAVVVifkwHc5sbbBH2zxSsZVj4JQnUVJCBqjTPToYk5eIoiGuIr7eCOPMC0cTExee1v16KN4Kw7bjoF1qvQoQ4hR%2BIMdGgEW%2BFTYtIGbAXg%2BEIhVVZOQW18BQfgclYdHOgK0b1R4hX2Y9RGoQE-jn8AeEhh5VSngUY8p5Z7YXns5BunEs4pUZp9Fm-1RAHGqCyO48ALAllgJVSQ7cpY30gfVVOddYGLwggg5%2BBRMCYwKJAMu7JRCsl8NYAA0ogRAWhz6NCEF4CkTJd5kyOBA2SpC55pwXouKhLxEFHEwMzBhFJSziHwAcfguMhLeHBFWRAv1UFSzLuVSqlZpI8igQpGBZ04FL1DgzBRKtlGMh3LSESWtoB-BYSCT88BfRDyNGJaApYmQj2IeIiaZD77pzSOLAGMxBFknwMIOYGgFhPwcbQp2DCdbxMpEyaAyT-DzEOAwzxLIdDwDOMLamoFaaqWXs3NwOB3ScO4bpbmADJgqIsGojRUZHzJypkjOptjZHCmzgo3ODDCGd0LIgXJks6jfi0DYS%2BI9jEV1MdVaeMk77TRTBxOx1DMmfFbgwphXjWGXysIFWi-ARHgMwpYmcfAoihH0PwQElUZBiJALEIAA)

#### Code

```fsharp
234<kg^(2/3(* foo *))>
```
            

#### Result

```fsharp
234<kg^(2/3) (* foo *) >

```
            
#### Problem description

Please describe here the Fantomas problem you encountered.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/main/CONTRIBUTING.md#bug-reports).

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.
- [ ] I would like a release if this problem is solved.

#### Options

Fantomas main branch at 2023-07-24T13:06:38Z - 0e07caba6301511d70a8187d5b6a1982d6c98638

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>
<sub>PS: It's unlikely that someone else will solve your specific issue, as it's something that you have a personal stake in.</sub>

### PR DESCRIPTION
**Pull request by dawedawe**
Fixes #2930
Fixes #2932 
Fixes #2933

### CODE REVIEW

File: src/Fantomas.Core/ASTTransformer.fs [hunk 1]
```diff
@@ -2016,16 +2016,39 @@ let mkSynRationalConst (creationAide: CreationAide) rc =
...
+                                                           divRange,
+                                                           denominator,
+                                                           denominatorRange,
+                                                           _),
+                                 range) ->
+            let openingParen =
+                let r =
+                    withEnd (Position.mkPos range.Start.Line (range.StartRange.StartColumn + 1)) range.StartRange
+
+                stn (creationAide.TextFromSource (fun () -> "(") r) r
```
Comment (nojaf) [inline src/Fantomas.Core/ASTTransformer.fs:2031]: Can this ever be anything else than `(`? It feels like overkill to grab this from the source text.

File: src/Fantomas.Core/ASTTransformer.fs [hunk 2]
```diff
@@ -2016,16 +2016,39 @@ let mkSynRationalConst (creationAide: CreationAide) rc =
...
+
             let n =
                 stn (creationAide.TextFromSource (fun () -> string numerator) numeratorRange) numeratorRange
 
+            let div = stn (creationAide.TextFromSource (fun () -> "/") divRange) divRange
```
Comment (nojaf) [inline src/Fantomas.Core/ASTTransformer.fs:2036]: Same remark here, will this ever be anything else than `/`?

**Review by nojaf — COMMENTED** [review]
Looks good, one open question left.

File: src/Fantomas.Core/ASTTransformer.fs (hunk 1 above)
Comment (dawedawe) [reply src/Fantomas.Core/ASTTransformer.fs:2031]: Currently it can only be `(`. I followed the idiom seen elsewhere to blindly. I'll adjust.

Comment (nojaf) [reply src/Fantomas.Core/ASTTransformer.fs:2031]: `creationAide.TextFromSource` is mostly used for user-defined code.
There are plenty of examples where we do `stn "x"` where `x` is a hardcoded thing.
https://github.com/fsprojects/fantomas/blob/646dcc9f9b28373247695abbbe79474819b80403/src/Fantomas.Core/ASTTransformer.fs#L74-L79

Comment (dawedawe) [reply src/Fantomas.Core/ASTTransformer.fs]: Yes, sorry. I looked at the wrong example.

**Comment by dawedawe** [comment]
> Great work!

Thanks, especially for the help and discussion regarding the AST change.

**Pull request merged**